example/lambdas/python-processing/pyproject.toml

```
@@ -0,0 +1,20 @@
+[project]
+name = "cumulus-python-processing"
+version = "21.0.1"
+description = "Python reference activity that utilizes the Cumulus Process SDK."
+requires-python = "~=3.12.0"
+
+dependencies = [
+    "boto3~=1.40.29",
```

I think probably what we really want here is:

```suggestion
    "boto3~=1.40",
```

The 'Compatible Release' version specifier is picky about what the following number means. Adding a patch release number means that only patch releases will be pulled in, backwards compatible minor version releases won't be allowed. 

https://peps.python.org/pep-0440/#compatible-release


Generally, I think it's good to make the accepted versions reasonably wide here so that new versions will continue to be pulled in even if the dependency bumps the minor version. Committing the `.lock` file takes care of guaranteeing reproducible builds.